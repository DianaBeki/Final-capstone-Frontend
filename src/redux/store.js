import { configureStore } from '@reduxjs/toolkit';
import doctorReducer from './doctor/doctorSlice';
import userSlice from './user/userSlice';

const store = configureStore({
  reducer: {
    doctor: doctorReducer,
    user: userSlice,
  },
});

export default store;
