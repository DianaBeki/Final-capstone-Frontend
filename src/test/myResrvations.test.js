import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { Provider } from 'react-redux';
import MyReservations from '../components/pages/myReservations';
import store from '../redux/store';

const render = (component) => rtlRender(
  <Provider store={store}>
    {component}
  </Provider>,
);

describe('MyReservations Component', () => {
  it('renders MyReservations component', () => {
    const { container } = render(<MyReservations />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
