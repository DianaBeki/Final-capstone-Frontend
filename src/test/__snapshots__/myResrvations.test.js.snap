// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyReservations Component renders MyReservations component 1`] = `
<div
  class="container mx-auto mt-8"
>
  <h2
    class="text-2xl font-bold mb-4 my-reserve-head"
  >
    My Reservations
  </h2>
  <table
    class="w-full border-collapse"
  >
    <thead>
      <tr
        class="bg-gray-200"
      >
        <th
          class="py-2 px-4 border"
        >
          Id
        </th>
        <th
          class="py-2 px-4 border"
        >
          Date
        </th>
        <th
          class="py-2 px-4 border"
        >
          City
        </th>
        <th
          class="py-2 px-4 border"
        >
          Doctor
        </th>
        <th
          class="py-2 px-4 border"
        >
          Actions
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
